### PR TITLE
upgrade github actions workflows

### PR DESCRIPTION
**Pull request by addyess**
* uses inclusive-workflow callable workflow
* updates `actions/checkout` and `actions/setup-python`
* corrects that each python matrix is tested
* uses setup-python's cache abilities
* runs the tests outside of tox on github actions to make use of the cached python requirements

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -24,13 +18,13 @@ jobs:
...
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - name: Cache pip
-        uses: actions/cache@v2
+        uses: actions/cache@v3
```
Comment (mateoflorido) [inline .github/workflows/test.yml:27]: What if we switch to the [cache property](https://github.com/actions/setup-python#caching-packages-dependencies) bundled in `setup-python`?

Comment (addyess) [reply .github/workflows/test.yml:27]: what's odd here is that we're running all the tests with tox... so all the caching in the world doesn't really help with the fact that tox goes and recreates it's own virtual environment.  It does beg the question though maybe the github actions could run without tox?

Comment (mateoflorido) [reply .github/workflows/test.yml:27]: You're right, caching doesn't really help. Concerning the running the Github actions without tox from my point of view, this change would not reduce our testing time enough to undertake those changes

Comment (addyess) [reply .github/workflows/test.yml]: let's try it anyway 👍🏻

Comment (addyess) [reply .github/workflows/test.yml]: @mateoflorido cool... please re-review.  Looks like this idea works

Comment (mateoflorido) [reply .github/workflows/test.yml]: Nice! LGTM, I'll rerun the tests to see if the pip cache kicks in

Comment (mateoflorido) [reply .github/workflows/test.yml]: Great, it looks like it's working and it's faster as well. 🥇

**Review by mateoflorido — APPROVED** [review]
LGTM! 🥇

**Pull request merged**